Objects/unicodeobject.c

```diff
@@ -6086,17 +6082,13 @@ _PyUnicode_DecodeUnicodeEscape(const char *s,
 
       error:
         endinpos = s-starts;
-        writer.min_length = end - s + writer.pos;
         if (unicode_decode_call_errorhandler_writer(
                 errors, &errorHandler,
                 "unicodeescape", message,
                 &starts, &end, &startinpos, &endinpos, &exc, &s,
                 &writer)) {
             goto onError;
         }
-        if (_PyUnicodeWriter_Prepare(&writer, writer.min_length, 127) < 0) {
-            goto onError;
```

Comment:
This is even more important. Since `WRITE_CHAR` doesn't check the size of the output buffer, we need to allocate the space for `writer.min_length = end - s + writer.pos` characters past the last written character.